features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/attachments/preview/AttachmentsPreviewPresenter.kt

```diff
@@ -81,10 +80,26 @@ class AttachmentsPreviewPresenter @AssistedInject constructor(
 
     private suspend fun sendMedia(
         mediaAttachment: Attachment.Media,
-        sendActionState: MutableState<Async<Unit>>,
+        sendActionState: MutableState<SendActionState>,
     ) {
-        sendActionState.runUpdatingState {
-            mediaSender.sendMedia(mediaAttachment.localMedia.uri, mediaAttachment.localMedia.info.mimeType, mediaAttachment.compressIfPossible)
+        val progressCallback = object : ProgressCallback {
+            override fun onProgress(current: Long, total: Long) {
+                sendActionState.value = SendActionState.Sending.Uploading(current.toFloat() / total)
```

Comment:
This works fine, but maybe we could also add `toFloat()` to `total`?

Reply:
As you want, it doesn't change anything